src/main/java/de/caritas/cob/messageservice/filter/SubdomainExtractor.java

```diff
@@ -0,0 +1,41 @@
+package de.caritas.cob.messageservice.filter;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.springframework.web.context.request.RequestContextHolder.currentRequestAttributes;
+
+import com.google.common.net.InternetDomainName;
+import de.caritas.cob.messageservice.api.helper.HttpUrlUtils;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+@Component
+@Setter
+@Slf4j
+@AllArgsConstructor
+public class SubdomainExtractor {
```

Comment:
unit tests should be moved to this service as well. It applies to all classes that have been taken from other microservice.

Reply:
done